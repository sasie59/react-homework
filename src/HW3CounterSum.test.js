import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW3CounterSum';
test('should HW3CounterSum', () => {
  const {container } =render(<HW />);
  expect(container).toMatchSnapshot;
});

test('從1開始累加至按的次數 ', () => {
  const {container} = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('從1開始累加至按的次數 ', () => {
  const {container} = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('從1開始累加至按的次數 ', () => {
  const {container} = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

