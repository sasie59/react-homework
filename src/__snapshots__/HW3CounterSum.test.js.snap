// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`從1開始累加至按的次數  1`] = `
<div>
  <div>
    <h1>
      HW3CounterSum
    </h1>
    <button>
      add
    </button>
    <br />
    0
  </div>
</div>
`;

exports[`從1開始累加至按的次數  2`] = `
<div>
  <div>
    <h1>
      HW3CounterSum
    </h1>
    <button>
      add
    </button>
    <br />
    3
  </div>
</div>
`;

exports[`從1開始累加至按的次數  3`] = `
<div>
  <div>
    <h1>
      HW3CounterSum
    </h1>
    <button>
      add
    </button>
    <br />
    10
  </div>
</div>
`;
